python/ray/data/_internal/planner/plan_read_op.py

```diff
@@ -36,6 +38,37 @@ def cleaned_metadata(read_task):
     return block_meta
 
 
+def _splitrange(n, k):
+    """Calculates array lens of np.array_split().
+
+    This is the equivalent of
+    `[len(x) for x in np.array_split(range(n), k)]`.
+    """
+    base = n // k
+    output = [base] * k
+    rem = n - sum(output)
+    for i in range(len(output)):
+        if rem > 0:
+            output[i] += 1
+            rem -= 1
+    assert rem == 0, (rem, output, n, k)
+    assert sum(output) == n, (output, n, k)
+    return output
+
+
+def _do_additional_splits(
+    blocks: Iterable[Block], _: TaskContext, additional_output_splits: int
+) -> Iterable[Block]:
+    assert additional_output_splits > 1
```

Comment:
the original code has `if additional_output_splits: ... else: ...` instead of assertion. Just to make sure nothing would be failed here. Look like `additional_output_splits` should always be larger than 1 - https://github.com/ray-project/ray/blob/master/python/ray/data/read_api.py#L402 . But please let me know if I am missing any other cases.

Reply:
Right, it should be always greater than 1. The original code has the if-else because it's also used for the no-additional-split code path.